pysat/tests/classes/cls_instrument_access.py

```diff
@@ -0,0 +1,1003 @@
+        return
+
+    def test_setting_partial_data(self):
+        """Test setting partial data by index."""
+
+        self.testInst.load(self.ref_time.year, self.ref_doy)
+        self.out = self.testInst
+        if self.testInst.pandas_format:
+            self.testInst[0:3] = 0
+            # First three values should be changed.
+            assert np.all(self.testInst[0:3] == 0)
+            # Other data should be unchanged.
+            assert np.all(self.testInst[3:] == self.out[3:])
+        else:
+            pytest.skip("This notation does not make sense for xarray")
+        return
+
+    @pytest.mark.parametrize("changed,fixed",
+                             [(0, slice(1, None)),
+                              ([0, 1, 2, 3], slice(4, None)),
+                              (slice(0, 10), slice(10, None)),
+                              (np.array([0, 1, 2, 3]), slice(4, None)),
+                              (dt.datetime(2009, 1, 1), slice(1, None)),
+                              (slice(dt.datetime(2009, 1, 1),
+                                     dt.datetime(2009, 1, 1, 0, 1)),
+                               slice(dt.datetime(2009, 1, 1, 0, 1), None))])
+    def test_setting_partial_data_by_inputs(self, changed, fixed):
+        """Check that data can be set using each supported index type.
+
+        Parameters
+        ----------
+        changed : index-like parameters
+            Index of values that change during the test.
+        fixed : index-like parameters
+            Index of values that should remain the same during the test.
+
+        """
+
+        self.testInst.load(self.ref_time.year, self.ref_doy)
+        self.testInst['doubleMLT'] = 2. * self.testInst['mlt']
+        self.testInst[changed, 'doubleMLT'] = 0
+        assert (self.testInst[fixed, 'doubleMLT']
+                == 2. * self.testInst[fixed, 'mlt']).all
+        assert (self.testInst[changed, 'doubleMLT'] == 0).all
+        return
+
+    def test_modifying_data_inplace(self):
+        """Test modification of data inplace."""
+
+        self.testInst.load(self.ref_time.year, self.ref_doy)
+        self.testInst['doubleMLT'] = 2. * self.testInst['mlt']
+        self.testInst['doubleMLT'] += 100
+        assert (self.testInst['doubleMLT']
+                == 2. * self.testInst['mlt'] + 100).all
+        return
+
+    @pytest.mark.parametrize("index", [([0, 1, 2, 3, 4]),
+                                       (np.array([0, 1, 2, 3, 4]))])
+    def test_getting_all_data_by_index(self, index):
+        """Test getting all data by index.
+
+        Parameters
+        ----------
+        index : index-like parameters
+            Index of values to retrieve.
+
+        """
+
+        self.testInst.load(self.ref_time.year, self.ref_doy)
+        a = self.testInst[index]
+        if self.testInst.pandas_format:
+            assert len(a) == len(index)
+        else:
+            assert a.sizes[self.xarray_epoch_name] == len(index)
```

Comment:
`a` should be changed to be a descriptive variable name.

Reply:
updating to `inst_subset`